packages/auth/src/Domain/Crypter/Crypter.ts

```diff
@@ -0,0 +1,43 @@
+import { CrypterInterface } from './CrypterInterface'
+import { SnCryptoNode } from '@standardnotes/sncrypto-node'
+import { createHash, randomBytes } from 'crypto'
+import { HexString } from '@standardnotes/sncrypto-common'
+
+export class Crypter implements CrypterInterface {
+  private readonly IV_BYTES = 16
+
+  /**
+   * @param crypto SnCryptoNode instance
+   * @param hexIv must be a 16-byte hex string suitable for AES 256 GCM initialization vector.
+   */
+  constructor(
+    private crypto: SnCryptoNode,
+    private hexIv?: HexString
+  ) {
+    if (this.hexIv === undefined) {
+      this.hexIv = randomBytes(this.IV_BYTES).toString('hex')
+    }
+  }
+
+  async encrypt(plaintext: string, secretKey: string): Promise<string> {
```

Comment:
I believe we can pull these in from sncrypto-common?
```suggestion
  async encrypt(plaintext: Utf8String, secretKey: Utf8String): Promise<string> {
```